3scale/tasks/main.yml

```diff
@@ -0,0 +1,172 @@
+---
+- name: Setting Up Results Var
+  ansible.builtin.set_fact:
+    test_results: []
+- name: Create Secret in target_namespace
+  block:
+    - name: Create Secret in target_namespace
+      kubernetes.core.k8s:
+        state: present
+        namespace: "{{ target_namespace }}"
+        definition:
+          apiVersion: v1
+          kind: Secret
+          metadata:
+            name: aws-auth
+          stringData:
+            AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID}
+            AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}
+            AWS_BUCKET: ${AWS_BUCKET}
+            AWS_REGION: ${AWS_REGION}
+          type: Opaque
+    - name: Create APIManager in target_namespace
+      kubernetes.core.k8s:
+        state: present
+        namespace: "{{ target_namespace }}"
+        definition:
+          apiVersion: apps.3scale.net/v1alpha1
+          kind: APIManager
+          metadata:
+            name: apimanager-sample
+          spec:
+            wildcardDomain: 3scale.${CLUSTER_BASE_URL}
+            system:
+              fileStorage:
+                simpleStorageService:
+                  configurationSecretRef:
+                    name: aws-auth
+    - name: Install and report - APIManager
+      kubernetes.core.k8s_info:
```

Comment:
There is no need for this additional task. You can put all of the register and wait* stuff in the previous task. It will give you back the final version of the resource in the var.